### PR TITLE
Modernize accessibility app

### PR DESCRIPTION
**Pull request by CarlSchwan**
- Use php 7.4 typed properties
- More typed methods (and replace existing php doc in some cases)

## Test plans

1. unit tests are working
2. Using the accessibility settings it also works

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

### CODE REVIEW

File: apps/accessibility/lib/AccessibilityProvider.php [hunk 1]
```diff
@@ -33,21 +33,12 @@
...
-	private $urlGenerator;
-
-	/** @var IL10N */
-	private $l;
+	protected string $appName;
+	private IURLGenerator $urlGenerator;
+	private IL10N $l;
 
 	/**
 	 * Account constructor.
```
Comment (juliushaertl) [inline apps/accessibility/lib/AccessibilityProvider.php:41]: I'd drop that as well, seems like a copied constructor default comment.

**Review by juliusknorr — APPROVED** [review]

File: apps/accessibility/lib/AccessibilityProvider.php [hunk 2]
```diff
@@ -69,6 +60,9 @@ public function getThemes() {
 		];
 	}
 
+	/**
+	 * @psalm-return array{id: string, img: string, title: string, enableLabel: string, text: string}
+	 */
 	public function getHighContrast() {
```
Comment (come-nc) [inline apps/accessibility/lib/AccessibilityProvider.php:66]:
```suggestion
	public function getHighContrast(): array {
```

File: apps/accessibility/lib/AccessibilityProvider.php [hunk 3]
```diff
@@ -79,6 +73,9 @@ public function getHighContrast() {
 		];
 	}
 
+	/**
+	 * @psalm-return array<array-key, array{id: string, img: string, title: string, enableLabel: string, text: string}>
+	 */
 	public function getFonts() {
```
Comment (come-nc) [inline apps/accessibility/lib/AccessibilityProvider.php:79]:
```suggestion
	public function getFonts(): array {
```

File: apps/accessibility/lib/Controller/AccessibilityController.php [hunk 4]
```diff
@@ -48,37 +48,20 @@
...
-	private $injectedVariables;
-
-	/** @var string */
-	private $appRoot;
-
-	public function __construct(string $appName,
+	/**
+	 * @param string $appName
+	 */
+	public function __construct($appName,
```
Comment (come-nc) [inline apps/accessibility/lib/Controller/AccessibilityController.php:64]:
```suggestion
	public function __construct(string $appName,
```

**Review by come-nc — CHANGES_REQUESTED** [review]

File: apps/accessibility/lib/Controller/AccessibilityController.php [hunk 5]
```diff
@@ -48,36 +48,19 @@
...
+	protected ITimeFactory $timeFactory;
+	private IUserSession $userSession;
+	protected IconsCacher $iconsCacher;
+	private \OC_Defaults $defaults;
+	private ?string $injectedVariables = null;
+	private string $appRoot;
 
+	/**
+	 * @param string $appName
+	 */
```
Comment (come-nc) [inline apps/accessibility/lib/Controller/AccessibilityController.php:63]:
```suggestion
```

**Review by come-nc — APPROVED** [review]
Remove leftover docblock and good to merge.

**Pull request merged**